src/mgr/BaseMgrModule.cc

```diff
@@ -92,22 +92,18 @@ class MonCommandCompletion : public Context
 
       auto set_fn = PyObject_GetAttrString(python_completion, "complete");
       ceph_assert(set_fn != nullptr);
-
-      auto pyR = PyLong_FromLong(r);
       auto pyOutBl = PyUnicode_FromString(outbl.to_str().c_str());
       auto pyOutS = PyUnicode_FromString(outs.c_str());
-      auto args = PyTuple_Pack(3, pyR, pyOutBl, pyOutS);
-      Py_DECREF(pyR);
+      auto rtn = PyObject_CallFunction(set_fn, "(iss)", r, pyOutBl, pyOutS);
```

Comment:
>  this is not the best practice of using PyObject_CallFunction(), if we are aiming at more efficient calling into Python methods, please use C arguments. 

please reconsider this comment.

> The 'finish' function previously retained memory, leading to extended holding periods by the garbage collector.

and, again the commit message is confusing. your change has nothing to do with the life cycle of the "finish" function.

Reply:
> > this is not the best practice of using PyObject_CallFunction(), if we are aiming at more efficient calling into Python methods, please use C arguments.
> 
> please reconsider this comment.

Done, i'll pass the c_str and will skip the object build



> and, again the commit message is confusing. your change has nothing to do with the life cycle of the "finish" function.

fixed.

@tchaikov i also added a change inside rest module to release some memory instead of holding results in a array without boundaries.